fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchIncludeTest.java

```diff
@@ -0,0 +1,585 @@
+        Response response =
+                target.path("Patient")
+                .queryParam("_include", "Patient:badParm")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Undefined Inclusion Parameter: Patient:badParm");
+    }
+
+    @Test(groups = { "server-search-include" })
+    public void testSearchIncludeWithSearchParameterTypeError() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_include", "Patient:address")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.BAD_REQUEST.getStatusCode());
+        assertExceptionOperationOutcome(response.readEntity(OperationOutcome.class),
+                "Inclusion Parameter must be of type 'reference'. The passed Inclusion Parameter is of type 'string': Patient:address");
+    }
+
+    @Test(groups = { "server-search-include" }, dependsOnMethods = {"testCreatePatient2"})
+    public void testSearchIncludeSingleIncludedResult() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_tag", tag)
+                .queryParam("gender", "female")
+                .queryParam("_include", "Patient:organization")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+
+        assertNotNull(bundle);
+        assertEquals(2, bundle.getEntry().size());
+        assertEquals(patient2Id, bundle.getEntry().get(0).getResource().getId());
+        assertEquals(SearchEntryMode.MATCH, bundle.getEntry().get(0).getSearch().getMode());
+        assertEquals(organization1Id, bundle.getEntry().get(1).getResource().getId());
+        assertEquals(SearchEntryMode.INCLUDE, bundle.getEntry().get(1).getSearch().getMode());
+    }
+
+    @Test(groups = { "server-search-include" }, dependsOnMethods = {"testCreatePatient2"})
+    public void testSearchIncludeSingleIncludedResultSameType() {
+        WebTarget target = getWebTarget();
+        Response response =
+                target.path("Patient")
+                .queryParam("_tag", tag)
+                .queryParam("gender", "female")
+                .queryParam("_include", "Patient:link")
+                .request(FHIRMediaType.APPLICATION_FHIR_JSON)
+                .get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Bundle bundle = response.readEntity(Bundle.class);
+
+        assertNotNull(bundle);
+        assertEquals(2, bundle.getEntry().size());
```

Comment:
This is pretty locked in to a specific result.  If there is bad data, is this test resilient?

Reply:
See above comment. I believe the tests are resilient based on scoping of the searches to only those resources created in the current test run.